src/ejabberd_c2s.erl

```diff
@@ -2349,22 +2349,25 @@ get_priority_from_presence(PresencePacket) ->
                          StateData :: state()) -> {mongoose_acc:t(), state()}.
 process_privacy_iq(Acc0, To, StateData) ->
     Acc = mongoose_acc:require(iq_query_info, Acc0),
-    IQ = mongoose_acc:get(iq_query_info, Acc),
-    Acc1 = mongoose_acc:put(iq, IQ, Acc),
-    From = mongoose_acc:get(from_jid, Acc1),
-    #iq{type = Type, sub_el = SubEl} = IQ,
-    {Acc2, NewStateData} = process_privacy_iq(Acc1, Type, To, StateData),
-    Res = mongoose_acc:get(iq_result, Acc2, {error, ?ERR_FEATURE_NOT_IMPLEMENTED}),
-    IQRes = case Res of
-                {result, Result} ->
-                    IQ#iq{type = result, sub_el = Result};
-                {result, Result, _} ->
-                    IQ#iq{type = result, sub_el = Result};
-                {error, Error} ->
-                    IQ#iq{type = error, sub_el = [SubEl, Error]}
-            end,
-    Acc3 = ejabberd_router:route(To, From, Acc2, jlib:iq_to_xml(IQRes)),
-    {Acc3, NewStateData}.
+    case mongoose_acc:get(iq_query_info, Acc) of
```

Comment:
can we use a separate function instead of case?

Reply:
We could, it's a tradeoff between large case and adding yet another function to `ejabberd_c2s` (and coming up with a meaningful name).

Reply:
Either way works for me, the problem I have would be to come up with a meaningful function name.